jobs/payment-jobs/tasks/electronic_funds_transfer_task.py

```diff
@@ -12,273 +12,144 @@
+        credit_invoice_links = cls._get_eft_credit_invoice_links_by_status(EFTCreditInvoiceStatus.PENDING_REFUND.value)
+        for invoice, credit_invoice_link, cfs_account in credit_invoice_links:
             try:
-                current_app.logger.debug(f'Unlinking Electronic Funds Transfer: {eft_short_name.id}')
-                payment_account: PaymentAccountModel = PaymentAccountModel.find_by_auth_account_id(
-                    eft_short_name.auth_account_id)
-                cfs_account = CfsAccountModel.find_effective_or_latest_by_payment_method(payment_account.id,
-                                                                                         PaymentMethod.EFT.value)
-                eft_credit = EFTCreditModel.find_by_payment_account_id(payment_account.id)
-
-                payment = db.session.query(PaymentModel) \
-                    .join(PaymentAccountModel, PaymentAccountModel.id == PaymentModel.payment_account_id) \
-                    .join(EFTCreditModel, EFTCreditModel.payment_account_id == PaymentAccountModel.id) \
-                    .join(EFTShortNamesModel, EFTShortNamesModel.id == EFTCreditModel.short_name_id) \
-                    .filter(EFTShortNamesModel.id == eft_short_name.id).first()
-
-                eft_short_name.version += 1
-
-                receipt_number = f'{payment.id}R{eft_short_name.version}'
-
-                CFSService.reverse_rs_receipt_in_cfs(cfs_account, receipt_number, ReverseOperation.CORRECTION.value)
-
-                CFSService.create_cfs_receipt(
-                    cfs_account=cfs_account,
-                    rcpt_number=f'R{receipt_number}',
-                    rcpt_date=payment.payment_date.strftime('%Y-%m-%d'),
-                    amount=payment.invoice_amount,
-                    payment_method=payment_account.payment_method,
-                    access_token=CFSService.get_token(PaymentSystem.FAS).json().get('access_token'))
-
-                cls._reset_invoices_and_references_to_created_for_electronic_funds_transfer(eft_short_name)
-
-                cls._apply_electronic_funds_transfers_to_pending_invoices(eft_short_name, receipt_number)
-
-                eft_credit.remaining_amount = eft_credit.amount
-
-                eft_short_name.save()
-
+                current_app.logger.debug(f'Reverse EFT Invoice Links in CFS - PayAccount: {invoice.payment_account_id}'
+                                         f'Id: {credit_invoice_link.id} - Amount: {credit_invoice_link.amount}'
+                                         f'- Invoice Id: {invoice.id}')
+                receipt_number = f'EFT-CIL-{credit_invoice_link.id}'
+                invoice_reference = InvoiceReferenceModel.find_by_invoice_id_and_status(
+                    invoice.id, InvoiceReferenceStatus.COMPLETED.value
+                )
+                # TODO need to handle reversal for disbursement, we can look for EFT CREATED invoices that are COMPLETED
```

Comment:
Disbursement todo

Reply:
We'll put in a row in that table and drive the EFT reversals from that table, so it's historical and straight forward and clear, no guessing.